examples/Bickley_jet_shallow_water.jl

```diff
@@ -0,0 +1,270 @@
+# the evolution of an unstable, geostrophically balanced, Bickley jet.
+# The model solves the governing equations for the shallow water model in
+# conservation form.  The geometry is that of a periodic channel
+# in the ``x``-direction with a flat bottom and a free-surface.  The initial
+# conditions are that of a Bickley jet with small amplitude perturbations.
+# The interested reader can see ["The nonlinear evolution of barotropically unstable jets," J. Phys. Oceanogr. (2003)](https://doi.org/10.1175/1520-0485(2003)033<2173:TNEOBU>2.0.CO;2)
+# for more details on this specific problem. 
+#
+# Unlike the other models, the fields that are simulated are the mass transports, 
+# ``uh`` and ``vh`` in the ``x`` and ``y`` directions, respectively,
+# and the height ``h``.  Note that ``u`` and ``v`` are the velocities, which 
+# can easily be computed when needed.
+#
+# ## Install dependencies
+#
+# First we make sure that we have all of the packages that are required to
+# run the simulation.
+# ```julia
+# using Pkg
+# pkg"add Oceananigans, NCDatasets, Plots, Printf, Polynomials"
+# ```
+#
+
+using Oceananigans
+using Oceananigans.Models: ShallowWaterModel
+
+# ## Two-dimensional domain 
+#
+# The shallow water model is a two-dimensional model and thus the number of vertical
+# points `N_z` must be set to one.  Note that ``Lz`` is the mean depth of the fluid. 
+
+Lx = 2π
+Ly = 20
+Lz = 1
+Nx = 128
+Ny = Nx
+
+grid = RegularRectilinearGrid(size = (Nx, Ny, 1),
+                            x = (0, Lx), y = (-Ly/2, Ly/2), z = (0, Lz),
+                            topology = (Periodic, Bounded, Bounded))
+
+# ## Physical parameters
+#
+# This is a toy problem and we choose the parameters so this jet idealizes
+# a relatively narrow mesoscale jet.   
+# The physical parameters are
+#
+#   * ``f``: Coriolis parameter
+#   * ``g``: Acceleration due to gravity
+#   * ``U``: Maximum jet speed
+#   * ``\Delta\eta``: Maximum free-surface deformation that is dictated by geostrophy
+#   * ``\epsilon`` : Amplitude of the perturbation
+
+f = 1
+g = 9.80665
+U = 1.0
+Δη = f * U / g
+ϵ = 1e-4
+nothing # hide
+
+# ## Building a `ShallowWaterModel`
+#
+# We use `grid`, `coriolis` and `gravitational_acceleration` to build the model.
+# Furthermore, we specify `RungeKutta3` for time-stepping and `WENO5` for advection.
+
+model = ShallowWaterModel(
+    timestepper=:RungeKutta3,
+    advection=WENO5(),
+    grid=grid,
+    gravitational_acceleration=g,
+    coriolis=FPlane(f=f),
+    )
+
+# ## Background state and perturbation
+#
+# The background velocity ``\overline u``, free-surface ``\eta`` and vorticity ``\Omega``
+#  are choosen to be a geostrophically balanced Bickely jet with maximum speed of ``U`` and 
+# maximum free-surface deformation of ``Δη``,
+#
+# ```math
+# \overline \eta(y) = - Δη \tanh(y),
+# ```
+# ```math
+# \overline    u(y) =    U \sech^2(y),
+# ```
+# ```math
+# \overilne    Ω(y) =  2 U \sech^2(y) \tanh(y).
+# ```
```

Comment:
```suggestion
# The background velocity ``\overline{u}`` and free-surface ``\overline{\eta}`` are chosen to 
# represent a geostrophically balanced Bickely jet with maximum speed of ``U`` and 
# maximum free-surface deformation of ``Δη``, i.e.,
#
# ```math
# \begin{align}
# \overline{\eta}(y) & = - Δη \tanh(y),
# \overline{u}(y) & = U \sech^2(y),
# \end{align}
# ```
# 
# The vorticity of the background state is ``Ω = - \partial_y u = 2 U \sech^2(y) \tanh(y)``.
```

Reply:
I simplified and also tried to fix it because it was not latexing properly.

(Note that you can just click the "commit suggestion" and it goes into the PR I think...)

Reply:
Good to know but will continue making changes manually this time, so I go through the motions.  Thanks

Reply:
I think `\sech` might need a special package (e.g., amsmath or what not) so I changed to `\mathrm{sech}`.